File: bun-resend-examples/javascript/examples/api-keys.js
import { Resend } from "resend";

const resend = new Resend(process.env.RESEND_API_KEY);

// 1. Create an API key
console.log("=== Creating API Key ===");
const { data: created, error: createError } = await resend.apiKeys.create({
  name: "Production Sending Key",
  permission: "sending_access",
});

if (createError) {
  console.error("Error creating API key:", createError);
  process.exit(1);
}
console.log("API key created:", created?.id);
console.log("Token (shown once, store it securely):", created?.token);

// 2. Rename the API key
// Only `name` is patchable - a leaked key must not be able to widen its own scope.
console.log("\n=== Renaming API Key ===");
const { error: updateError } = await resend.apiKeys.update(created.id, {
  name: "Production Sending Key v2",
});

if (updateError) {
  console.error("Error renaming API key:", updateError);
  process.exit(1);
}
console.log("API key renamed: Production Sending Key -> Production Sending Key v2");

// 3. List API keys
console.log("\n=== Listing API Keys ===");
const { data: keys } = await resend.apiKeys.list();
keys?.data.forEach((key) => {
  console.log(`  - ${key.name} (${key.id})`);
});

// 4. Delete the API key
console.log("\n=== Deleting API Key ===");
await resend.apiKeys.remove(created.id);
console.log("API key deleted:", created?.id);

console.log("\nDone! Full API key lifecycle complete.");

File: dotnet-resend-examples/Examples/ApiKeys.cs
namespace ResendExamples;

using Resend;

public static class ApiKeys
{
    public static async Task RunAsync()
    {
        var apiKey = Environment.GetEnvironmentVariable("RESEND_API_KEY")
            ?? throw new Exception("RESEND_API_KEY environment variable is required");

        var client = ResendClient.Create(apiKey);

        // 1. Create a new API key
        Console.WriteLine("=== Creating API Key ===");
        var created = await client.ApiKeyCreateAsync("Example Key");
        var apiKeyId = created.Content.Id;
        Console.WriteLine($"Created key: {apiKeyId}");
        Console.WriteLine($"Token (shown only once): {created.Content.Token}");

        // 2. Rename the API key
        // Only the name can be changed here; permission and domain restriction are
        // fixed at creation time, so a leaked key can't be used to widen its own scope.
        Console.WriteLine("\n=== Renaming API Key ===");
        await client.ApiKeyUpdateAsync(apiKeyId, "Example Key (renamed)");
        Console.WriteLine("Key renamed: Example Key -> Example Key (renamed)");

        // 3. List all API keys
        Console.WriteLine("\n=== Listing API Keys ===");
        var keys = await client.ApiKeyListAsync();
        Console.WriteLine($"Found {keys.Content.Count} key(s)");

        foreach (var key in keys.Content)
        {
            Console.WriteLine($"  - {key.Name} (id: {key.Id}, created: {key.MomentCreated})");
        }

        // 4. Delete the API key
        Console.WriteLine("\n=== Deleting API Key ===");
        await client.ApiKeyDeleteAsync(apiKeyId);
        Console.WriteLine($"Deleted key: {apiKeyId}");

        Console.WriteLine("\nDone!");
    }
}
